eyelid-match-ops/src/primitives/poly/test/inv.rs

```diff
@@ -0,0 +1,29 @@
+//! Tests for polynomial inverse.
+
+use super::super::*;
+use rand::Rng;
+
+/// This sampling is similar to what will be necessary for YASHE KeyGen
+/// TODO: generate Gaussian distribution instead of "uniform"
+pub fn sample() -> Poly {
+    let mut rng = rand::thread_rng();
+    let mut res = Poly::zero();
+    let max_coeff = 8;
+    let t = 2;
+    for i in 0..MAX_POLY_DEGREE {
+        let coeff_rand = rng.gen_range(1..max_coeff);
+        res[i] = Coeff::from(t * coeff_rand);
+    }
+    res[0] += Coeff::one();
+    res
+}
+
+#[test]
+fn test_inverse() {
+    let f = sample();
```

Comment:
Maybe we could consider using [proptests](https://proptest-rs.github.io/proptest/intro.html).

Very nice suggestions!

Reply:
> Maybe we could consider using [proptests](https://proptest-rs.github.io/proptest/intro.html).

Sounds good, did you want to open a ticket?

Reply:
[Here](https://github.com/Inversed-Tech/eyelid/issues/55) it is.

Reply:
Thanks! I made a comment on that ticket about how `proptest` struggles to find rare edge cases.

Are you ok with adding the list of tests above to this PR?
Let me know if you need any help.

Reply:
Done

Reply:
I don’t think these tests can be done with proptests:
- [ ] Test standard known inverses, for example, that -1 and 1 are their own inverses, and 0 doesn’t have an inverse.
- [ ] If there’s a polynomial we know the inverse of, that would also be useful to test. If not, we could generate a test vector and its inverse, and check that.

So they either need a separate ticket, or they could be added to this PR. I don’t mind merging this PR without those tests, but I’d like to have tests merged soon after PR, to avoid bugs.

As part of the fixed test vectors, it might be useful to test both representations of -1 (`COEFF_MODULUS - 1` and `X^N`), as well as both representations of zero.